tools/Azure.Mcp.Tools.Speech/src/Commands/Stt/SttRecognizeCommand.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Extensions;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.Speech.Models;
+using Azure.Mcp.Tools.Speech.Options;
+using Azure.Mcp.Tools.Speech.Options.Stt;
+using Azure.Mcp.Tools.Speech.Services;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.Speech.Commands.Stt;
+
+public sealed class SttRecognizeCommand(ILogger<SttRecognizeCommand> logger) : BaseSpeechCommand<SttRecognizeOptions>()
+{
+    internal record SttRecognizeCommandResult(ContinuousRecognitionResult Result);
+
+    private const string CommandTitle = "Recognize Speech from Audio File";
+    private readonly ILogger<SttRecognizeCommand> _logger = logger;
+
+    public override string Name => "recognize";
+
+    public override string Description =>
+        """
+        Recognize speech from an audio file using Azure AI Services Speech. This command takes an audio file and converts it to text using advanced speech recognition capabilities.
+        You must provide an Azure AI Services endpoint (e.g., https://your-service.cognitiveservices.azure.com/) and a path to the audio file.
+        Supported audio formats include WAV, MP3, OPUS/OGG, FLAC, ALAW, MULAW, MP4, M4A, and AAC. Compressed formats require GStreamer to be installed on the system.
+        Optional parameters include language specification, phrase hints for better accuracy, output format (simple or detailed), and profanity filtering.
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new()
+    {
+        Destructive = false,
+        Idempotent = true,
+        OpenWorld = false,
+        ReadOnly = true,
+        LocalRequired = true, // Requires local audio file access
+        Secret = false
+    };
+
+    protected override void RegisterOptions(Command command)
+    {
+        _logger.LogDebug("RegisterOptions");
+
+        base.RegisterOptions(command);
+
+        var fileOption = SpeechOptionDefinitions.File.AsRequired();
+        command.Options.Add(fileOption);
+        var languageOption = SpeechOptionDefinitions.Language.AsOptional();
+        command.Options.Add(languageOption);
+        var phrasesOption = SpeechOptionDefinitions.Phrases.AsOptional();
+        command.Options.Add(phrasesOption);
+        var formatOption = SpeechOptionDefinitions.Format.AsOptional();
+        command.Options.Add(formatOption);
+        var profanityOption = SpeechOptionDefinitions.Profanity.AsOptional();
+        command.Options.Add(profanityOption);
```

Comment:
Only use `AsRequired` and `AsOptional` when it will change the `Required` property of the Option